src/helics/application_api/Federate.cpp

```diff
@@ -834,12 +839,30 @@ void Federate::registerFilterInterfacesJson(const std::string& jsonString)
 
             auto opType = filterTypeFromString(operation);
             if ((useTypes) && (operation != "custom")) {
-                std::cerr << "input and output types may only be specified for custom filters\n";
+                if (strictConfigChecking) {
+                    logMessage(helics_log_level_error,
+                               "input and output types may only be specified for custom filters");
+                    throw(InvalidParameter(
+                        "input and output types may only be specified for custom filters"));
+                } else {
+                    logMessage(helics_log_level_warning,
+                               "input and output types may only be specified for custom filters");
+                }
                 continue;
             }
             if (!useTypes) {
                 if (opType == filter_types::unrecognized) {
-                    std::cerr << "unrecognized filter operation:" << operation << '\n';
+                    if (strictConfigChecking) {
```

Comment:
@trevorhardy this is the code that would catch the invalid filter type and generate an error